src/librados/PoolAsyncCompletionImpl.h

```diff
@@ -15,104 +15,107 @@
 #ifndef CEPH_LIBRADOS_POOLASYNCCOMPLETIONIMPL_H
 #define CEPH_LIBRADOS_POOLASYNCCOMPLETIONIMPL_H
 
-#include "common/Cond.h"
-#include "common/Mutex.h"
-#include "include/Context.h"
+#include <mutex>
+#include <condition_variable>
+
+#include <boost/intrusive_ptr.hpp>
+
+#include "common/ceph_mutex.h"
 #include "include/rados/librados.h"
 #include "include/rados/librados.hpp"
 
 namespace librados {
   struct PoolAsyncCompletionImpl {
-    Mutex lock;
-    Cond cond;
-    int ref, rval;
-    bool released;
-    bool done;
+    ceph::mutex lock = ceph::make_mutex("PoolAsyncCompletionImpl lock");
+    ceph::condition_variable cond;
+    int ref = 1;
+    int rval = 0;
+    bool released = false;
+    bool done = false;
 
-    rados_callback_t callback;
-    void *callback_arg;
+    rados_callback_t callback = nullptr;
+    void *callback_arg = nullptr;
 
-    PoolAsyncCompletionImpl() : lock("PoolAsyncCompletionImpl lock"),
-				ref(1), rval(0), released(false), done(false),
-				callback(0), callback_arg(0) {}
+    PoolAsyncCompletionImpl() = default;
 
     int set_callback(void *cb_arg, rados_callback_t cb) {
-      lock.Lock();
+      std::scoped_lock l(lock);
       callback = cb;
       callback_arg = cb_arg;
-      lock.Unlock();
       return 0;
     }
     int wait() {
-      lock.Lock();
+      std::unique_lock l(lock);
       while (!done)
-	cond.Wait(lock);
-      lock.Unlock();
+	cond.wait(l);
       return 0;
     }
     int is_complete() {
-      lock.Lock();
+      std::scoped_lock l(lock);
       int r = done;
-      lock.Unlock();
       return r;
     }
     int get_return_value() {
-      lock.Lock();
+      std::scoped_lock l(lock);
       int r = rval;
-      lock.Unlock();
       return r;
     }
     void get() {
-      lock.Lock();
+      std::scoped_lock l(lock);
       ceph_assert(ref > 0);
       ref++;
-      lock.Unlock();
     }
     void release() {
-      lock.Lock();
+      std::scoped_lock l(lock);
       ceph_assert(!released);
       released = true;
-      put_unlock();
     }
     void put() {
-      lock.Lock();
-      put_unlock();
-    }
-    void put_unlock() {
-      ceph_assert(ref > 0);
+      std::unique_lock l(lock);
+//      put_unlock();
       int n = --ref;
-      lock.Unlock();
+      l.unlock();
       if (!n)
 	delete this;
     }
+//    void put_unlock() {
```

Comment:
Dead?

Reply:
Ah, thank you for pointing that out. I wanted to make sure nobody was calling that before removing it and I forgot to remove it.

Reply:
Removed.